### DISCUSSION TITLE
Getting DB Constraint error

### DISCUSSION THREAD

**Discussion by Rathna-K** [teslamate-org/teslamate#2005] [Q&A] [answered]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

recently upgraded to edge and used recpatcha tools to login. the admin screen shows spinning wheel with the log below in docker. based on another issue here, i also trfied reindex commands and still no-go.

### Relevant log output

```shell
2021-09-28 10:13:11.035 car_id=1 [info] Start / :online
2021-09-28 10:13:11.059 [error] GenStateMachine :"1" terminating
** (Ecto.ConstraintError) constraint error when attempting to insert struct:

    * positions_pkey (unique_constraint)

If you would like to stop this constraint violation from raising an
exception and instead add it as an error to your changeset, please
call `unique_constraint/3` on your changeset with the constraint
`:name` as an option.

The changeset defined the following constraints:

    * positions_drive_id_fkey (foreign_key_constraint)
    * positions_car_id_fkey (foreign_key_constraint)

    (ecto 3.7.1) lib/ecto/repo/schema.ex:783: anonymous fn/4 in Ecto.Repo.Schema.constraints_to_errors/3
    (elixir 1.12.3) lib/enum.ex:1582: Enum."-map/2-lists^map/1-0-"/2
    (ecto 3.7.1) lib/ecto/repo/schema.ex:768: Ecto.Repo.Schema.constraints_to_errors/3
    (ecto 3.7.1) lib/ecto/repo/schema.ex:749: Ecto.Repo.Schema.apply/4
    (ecto 3.7.1) lib/ecto/repo/schema.ex:367: anonymous fn/15 in Ecto.Repo.Schema.do_insert/4
    (teslamate 1.25.0-dev) lib/teslamate/vehicles/vehicle.ex:698: anonymous fn/4 in TeslaMate.Vehicles.Vehicle.handle_event/4
    (ecto_sql 3.7.0) lib/ecto/adapters/sql.ex:1013: anonymous fn/3 in Ecto.Adapters.SQL.checkout_or_transaction/4
    (db_connection 2.4.0) lib/db_connection.ex:1512: DBConnection.run_transaction/4
```

### Type of installation

Docker

### Version

1.25.0-dev

**Comment by adriankumpf** [comment]
> based on another issue here, i also trfied reindex commands and still no-go.

Can you elaborate on the issue you had and which commands you ran in response?

Also, run these command to check if the table is somehow corrupted:

```sql
SELECT nextval('positions_id_seq');
SELECT max(id) from positions;
```

**Comment by Rathna-K** [comment]
this is what I looked up: https://github.com/adriankumpf/teslamate/issues/465
both the queries above returned a value and no errors. 
Also, I recently backed up the DB and upgraded postgres and restored the dump. not sure if that impacted anything as well.

**Reply by adriankumpf** [reply]
If the first value is lower than the second one there is in issue with the primary key sequence. According to [this SO answer](https://stackoverflow.com/questions/8745051/postgres-manually-alter-sequence/8745101#8745101) the following command should fix it:

```sql
SELECT setval('positions_id_seq', (SELECT max(id) from positions));
```

**Comment by adriankumpf** [comment, marked as answer]
I'll convert this to a discussion as there is no indication of a bug in TeslaMate but there seems to be some inconsistency in the database.

**Reply by Rathna-K** [reply]
Thank you Mr. Kumpf, you are the best!!
